### ISSUE TITLE
linter: [vscode] Failed to initialize oxlintrc config

### ISSUE THREAD

**Issue by magicdawn** [oxc-project/oxc#17384]
### What version of Oxlint are you using?

1.35.0

### What command did you run?

save changes

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": ["typescript", "unicorn", "oxc", "react"],
  "rules": {
    "no-unused-vars": "off",
    "no-unused-expressions": "off",
    "react/jsx-no-undef": "off",
    "react/jsx-no-target-blank": "off",
    "react/exhaustive-deps": "off",
    "react/rules-of-hooks": "error"
  }
}
```

### What happened?

in vscode with settings.json `oxc.configPath` set to empty string
```json
{
  "oxc.configPath": ""
}
```

will cause `WARN  oxc_language_server::linter::server_linter] Failed to initialize oxlintrc config`
hence oxc config not recognized.

I looked into the code and found
https://github.com/oxc-project/oxc/blob/oxlint_v1.35.0/crates/oxc_language_server/src/linter/server_linter.rs#L58

empty string is used and not fallback to default `.oxlintrc.json`, so resulting `Failed to initialize oxlintrc config` error

but I was not meant to config `oxc.configPath` to an empty string, it's auto generated by a click on vscode  settings UI
![Image](https://github.com/user-attachments/assets/3ac7eed6-787c-4d3a-b335-a5a06f493e74)

## Code

https://github.com/oxc-project/oxc/blob/oxlint_v1.35.0/crates/oxc_language_server/src/linter/server_linter.rs#L58

I confirmed the behavior with ChatGPT.
<img width="1101" height="1211" alt="Image" src="https://github.com/user-attachments/assets/a8bd6b1e-5d96-4040-b258-15a91679959b" />

empty string should use default fallback too:
```rust
let config_path = match options.config_path.as_deref() {
    Some("") | None => LINT_CONFIG_FILE,
    Some(v) => v,
};
```
